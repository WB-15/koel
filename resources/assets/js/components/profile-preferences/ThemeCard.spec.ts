import UnitTestCase from '@/__tests__/UnitTestCase'
import { expect, it } from 'vitest'
import { fireEvent } from '@testing-library/vue'
import ThemeCard from './ThemeCard.vue'

const theme: Theme = {
  id: 'sample',
  thumbnailColor: '#f00'
}

new class extends UnitTestCase {
  private renderComponent () {
    return this.render(ThemeCard, {
      props: {
        theme
      }
    })
  }

  protected test () {
    it('renders', () => {
      expect(this.renderComponent().html()).toMatchSnapshot()
    })

    it('emits an event when selected', async () => {
      const { emitted, getByTestId } = this.renderComponent()
      await fireEvent.click(getByTestId('theme-card-sample'))
      expect(emitted().selected[0]).toEqual([theme])
    })
  }
}
