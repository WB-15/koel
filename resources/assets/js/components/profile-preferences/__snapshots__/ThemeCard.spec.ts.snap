// Vitest Snapshot v1

exports[`renders 1`] = `
"<div class=\\"theme\\" data-testid=\\"theme-card-sample\\" style=\\"background-color: #f00;\\" title=\\"Set current them to Sample\\" role=\\"button\\" data-v-3bc20152=\\"\\">
  <div class=\\"name\\" data-v-3bc20152=\\"\\">Sample</div>
</div>"
`;
